kvmagent/kvmagent/plugins/vms/tpm.py

```
@@ -60,3 +67,83 @@ def check_tpm_state_vm_host_file_path_format(path):
+    return None
+
+
+def parse_vtpm_libvirt_secret_uuid_from_domain_xml(domain_xml):
+    """
+    Read libvirt secret UUID from domain XML for vTPM:
+    devices/tpm/backend/encryption/@secret
+    Parsed with xml.etree.ElementTree only (no regex on XML body).
+    """
+    if not domain_xml:
+        return None
+    tree = etree.ElementTree(etree.fromstring(domain_xml))
```

Ruff (0.15.10)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)